File: Debug/Barcode/Examples/VS2013/GizmoxVisualWebGui_VB/Form1.vb
Imports Gizmox.WebGUI.Common.Gateways
Imports Gizmox.WebGUI.Common.Interfaces

Public Class Form1
    Implements IGatewayControl

    Public Function GetGatewayHandler(ByVal objContext As IContext, ByVal strAction As String) As IGatewayHandler Implements IGatewayControl.GetGatewayHandler
        Select Case strAction
            Case "Barcode"
                Return New BarcodeHandler(Me._txtNumber.Text)
        End Select
        Return Nothing
    End Function

    Private Sub _btnRender_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _btnRender.Click
        Dim reference As New GatewayReference(Me, "Barcode")
        Me.HtmlBox1.Url = reference.ToString()
        Me.HtmlBox1.Update()
    End Sub

End Class

File: Debug/Barcode/Examples/VS2013/GizmoxVisualWebGui_CS/Form1.cs
using System;
using Gizmox.WebGUI.Common.Gateways;
using Gizmox.WebGUI.Common.Interfaces;
using Gizmox.WebGUI.Forms;

namespace GizmoxVisualWebGui_CS
{
    public partial class Form1 : Form, IGatewayControl
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        public IGatewayHandler GetGatewayHandler(IContext objContext, string strAction)
        {
            switch (strAction)
            {
                case "Barcode":
                    return new BarcodeHandler(this._txtBarcodeNumber.Text);
                default:
                    return null;
            }
        }

        private void button1_Click(object sender, EventArgs e)
        {
            GatewayReference reference = new GatewayReference(this, "Barcode");
            this.htmlBox1.Url = reference.ToString();
            this.htmlBox1.Update();
        }
    };
}